### ISSUE TITLE
Strange character being streamed on command line when running image classification benchmark (TFLite on RPi)

### ISSUE THREAD

**Issue by sindhujapr** [mlcommons/inference_results_v0.7#11]
I was following instructions from [here](https://github.com/mlperf/inference_results_v0.7/tree/master/closed/Dividiti/code/resnet50/image-classification-tflite-loadgen):
1. preprocessed on x86
2. transferred rgb8 images to RPi
3. detected processed images
4. ran benchmark using this command:

ck benchmark program:image-classification-tflite-loadgen \
--speed --repetitions=1 \
--env.CK_VERBOSE=1 \
--env.CK_LOADGEN_SCENARIO=SingleStream \
--env.CK_LOADGEN_MODE=PerformanceOnly \
--env.CK_LOADGEN_DATASET_SIZE=1024 \
--env.CK_LOADGEN_BUFFER_SIZE=1024 \
--dep_add_tags.weights=model,tflite,resnet \
--dep_add_tags.library=tflite,v1.15 \
--dep_add_tags.images=side.224,preprocessed \
--dep_add_tags.loadgen-config-file=image-classification-tflite \
--dep_add_tags.python=v3 \
--skip_print_timers

It starts to run and then starts printing a series of "1"s and "Qp"s like this:

![image](https://user-images.githubusercontent.com/9027766/99462946-bd727380-2902-11eb-8c86-f391e3112563.png)

Why are there these characters being printed? How can I move past them to get to the results?

Thank you for your time,
Sindhuja

**Comment by sindhujapr** [comment]
On doing a recursive search ( grep -rl Qp . ), it appears that all the preprocessed images have "Qp" and "1". Is this expected?

**Comment by psyhtest** [comment]
Hi @sindhujapr, absolutely nothing to worry about - it's just a progress report: `l` (not `1`) means "load", `Q` means "query", `p` means "prediction". You should observe exactly 1024 `Qp` [diphthongs](https://en.wikipedia.org/wiki/Diphthong), corresponding to the [minimum number of queries](https://github.com/mlperf/inference_policies/blob/master/inference_rules.adoc#3-scenarios) for the SingleStream scenario.

This can be disabled by passing `--env.CK_VERBOSE=0` instead of `--env.CK_VERBOSE=1`:

```bash
$ ck benchmark program:image-classification-tflite-loadgen \
--speed --repetitions=1 \
--env.CK_VERBOSE=0 \
--env.CK_LOADGEN_SCENARIO=SingleStream \
--env.CK_LOADGEN_MODE=PerformanceOnly \
--env.CK_LOADGEN_DATASET_SIZE=1024 \
--env.CK_LOADGEN_BUFFER_SIZE=1024 \
--dep_add_tags.weights=model,tflite,resnet \
--dep_add_tags.library=tflite,v1.15 \
--dep_add_tags.images=side.224,preprocessed \
--dep_add_tags.loadgen-config-file=image-classification-tflite \
--dep_add_tags.python=v3 \
--skip_print_timers
...
executing code ...
Graph file: /home/anton/CK_TOOLS/model-tflite-mlperf-resnet-no-argmax-downloaded/resnet50_v1.no-argmax.tflite
Image dir: /datasets/dataset-imagenet-preprocessed-using-opencv-crop.875-full-inter.linear-side.224
Image list: image_list.txt
Image size: 224
Image channels: 3
Prediction classes: 1000
Result dir: predictions
How many images fit in memory buffer: 1024
Normalize: 0
Subtract mean: 1
Per-channel means to subtract: 123.68, 116.78, 103.94
Number of available imagefiles: 1024

Loading graph...
Input tensor dimensions (NHWC): 1*224*224*3
Output tensor dimensions: 1*1001
Path to mlperf.conf : /home/anton/CK_TOOLS/mlperf-inference-r0.7/inference/mlperf.conf
Path to user.conf : user.conf
Model Name: resnet50
LoadGen Scenario: SingleStream
LoadGen Mode: PerformanceOnly
...
                        
------------------------------------------------------------                           
|            LATENCIES (in nanoseconds and fps)            |
------------------------------------------------------------                           
Number of queries run: 1024
Min latency:                      721793169ns  (1.38544 fps)
Median latency:                   733313874ns  (1.36367 fps)
Average latency:                  733938666ns  (1.36251 fps)
90 percentile latency:            740866704ns  (1.34977 fps)
Max latency:                      776810205ns  (1.28732 fps)
------------------------------------------------------------
```

The full performance run thus takes 12-13 minutes at the speed of 1.35 queries per second. I prefer to watch progress as it's much more fun than to stare at a frozen screen!

**Comment by sindhujapr** [comment]
Thank you very much for all this information @psyhtest - it makes a lot of sense now! I did observe 1024 Qp's and it was too good a number to be meaningless.
+1 to watching progress as opposed to staring and hoping everything is going well in the run.

I had 2 follow up questions:
1. How can I do the accuracy run using the above command? I tried:
    a. --env.CK_LOADGEN_MODE=Accuracy
    b. Not having the flag "--env.CK_LOADGEN_MODE"
    c. --env.CK_MODE=accuracy
but I ended up seeing the same results as the performance run - no specific field in the output for accuracy.

2. How can find/generate the result log files - mlperf_summary or mlperf_accuracy? Is there a flag to enable the creation of these files? I couldn't find them after the run.

I am happy to close this comment and open the above questions as a new issue too.

Thank you,
Sindhuja

**Comment by psyhtest** [comment]
Hi @sindhujapr, The implementation ([`program:image-classification-tflite-loadgen`](https://github.com/ctuning/ck-mlperf/tree/master/program/image-classification-tflite-loadgen)) always generates LoadGen log files. They can be found in the program's temporary directory after each run. 

After a performance run `mlperf_log_accuracy.json` contains an empty list, while `mlperf_log_summary.txt` is over a kilobyte e.g.:
```bash
$ ls -la `ck find program:image-classification-tflite-loadgen`/tmp/mlperf_log*
-rw-r--r-- 1 anton dvdt     4 Nov 25 09:26 /home/anton/CK_REPOS/ck-mlperf/program/image-classification-tflite-loadgen/tmp/mlperf_log_accuracy.json
-rw-r--r-- 1 anton dvdt 13455 Nov 25 09:26 /home/anton/CK_REPOS/ck-mlperf/program/image-classification-tflite-loadgen/tmp/mlperf_log_detail.txt
-rw-r--r-- 1 anton dvdt  1632 Nov 25 09:26 /home/anton/CK_REPOS/ck-mlperf/program/image-classification-tflite-loadgen/tmp/mlperf_log_summary.txt
-rw-r--r-- 1 anton dvdt     0 Nov 25 09:14 /home/anton/CK_REPOS/ck-mlperf/program/image-classification-tflite-loadgen/tmp/mlperf_log_trace.json
```

After an accuracy run (`--env.CK_LOADGEN_MODE=AccuracyOnly`), however, it's the other way round e.g.:
```bash
$ ck benchmark program:image-classification-tflite-loadgen \
--speed --repetitions=1 \
--env.CK_VERBOSE=1 \
--env.CK_LOADGEN_SCENARIO=SingleStream \
--env.CK_LOADGEN_MODE=AccuracyOnly \
--env.CK_LOADGEN_DATASET_SIZE=50 \
--env.CK_LOADGEN_BUFFER_SIZE=50 \
--dep_add_tags.weights=model,tflite,resnet \
--dep_add_tags.library=tflite,v1.15 \
--dep_add_tags.images=side.224,preprocessed \
--dep_add_tags.loadgen-config-file=image-classification-tflite \
--dep_add_tags.python=v3 \
--skip_print_timers

$ ls -la `ck find program:image-classification-tflite-loadgen`/tmp/mlperf_log*
-rw-r--r-- 1 anton dvdt 2783 Nov 27 16:09 /home/anton/CK_REPOS/ck-mlperf/program/image-classification-tflite-loadgen/tmp/mlperf_log_accuracy.json
-rw-r--r-- 1 anton dvdt 9435 Nov 27 16:09 /home/anton/CK_REPOS/ck-mlperf/program/image-classification-tflite-loadgen/tmp/mlperf_log_detail.txt
-rw-r--r-- 1 anton dvdt   74 Nov 27 16:09 /home/anton/CK_REPOS/ck-mlperf/program/image-classification-tflite-loadgen/tmp/mlperf_log_summary.txt
-rw-r--r-- 1 anton dvdt    0 Nov 27 16:09 /home/anton/CK_REPOS/ck-mlperf/program/image-classification-tflite-loadgen/tmp/mlperf_log_trace.json
```

**Comment by sindhujapr** [comment]
Great, thank you very much for these details! I will look at the tmp folders after using the above commands.

**Comment by psyhtest** [comment]
@sindhujapr Please note that we did not use the "classical" CK interface to run experiments for our v0.7 submission. Instead, we used the new "CmdGen" interface as detailed in our [measurements README](https://github.com/mlperf/inference_results_v0.7/blob/master/closed/Dividiti/measurements/xavier-tflite-v2.2.0-ruy/resnet50/singlestream/README.md). See examples of running on Xavier with TFLite v2.2 below.

### Performance

```
$ ck run cmdgen:benchmark.tflite-loadgen --library=tflite-v2.2.0-ruy \
--model=resnet50 --scenario=singlestream --mode=performance --target_latency=88 \
--verbose --sut=xavier
```

### Accuracy

```
$ ck run cmdgen:benchmark.tflite-loadgen --library=tflite-v2.2.0-ruy \
--model=resnet50 --scenario=singlestream --mode=accuracy --dataset_size=50000 \
--verbose --sut=xavier
...
accuracy=76.442%, good=38221, total=50000
```
